sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVInferSchema.scala

```diff
@@ -177,7 +191,8 @@ private[csv] object CSVTypeCast {
       datum: String,
       castType: DataType,
       nullable: Boolean = true,
-      nullValue: String = ""): Any = {
+      nullValue: String = "",
+      dateFormat: SimpleDateFormat = null): Any = {
```

Comment:
This will conflict with https://github.com/apache/spark/pull/11947 (this is making the last argument as a single option not for the multiple individual parameters like I did for `infer()` above).